persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -108,7 +110,52 @@ public static SyntaxTree createPesistToml(String name) {
         return SyntaxTree.from(textDocument);
     }
 
-    public static HashMap<String, String> readToml(Path configPath, String name) throws BalException {
+    public static HashMap<String, String> populateConfiguration(HashMap<String, String> persistConfig, Path configPath)
+            throws BalException {
+        Path fileNamePath = configPath.getFileName();
+        try {
+            TextDocument configDocument = TextDocuments.from(Files.readString(configPath));
+            SyntaxTree syntaxTree = SyntaxTree.from(configDocument, fileNamePath.toString());
+            DocumentNode rootNote = syntaxTree.rootNode();
+            NodeList<DocumentMemberDeclarationNode> nodeList = rootNote.members();
+            for (String configKey : persistConfig.keySet()) {
+                if (Pattern.matches(PLACEHOLDER_PATTERN, persistConfig.get(configKey))) {
+                    boolean configExists = false;
+                    String[] placeHolderValues = persistConfig.get(configKey).replaceAll("\"", "").split("\\.");
+                    String relatedKey =  placeHolderValues[placeHolderValues.length - 1].replaceAll("}", "");
+                    String placeHolderTable = persistConfig.get(configKey)
+                            .replaceAll("\\$\\{", "").replaceAll("}", "").replaceAll("\\." + relatedKey, "");
+                    for (Object member : nodeList) {
```

Comment:
Use specific type, `DocumentMemberDeclarationNode`

Reply:
updated in [0b4967f](https://github.com/ballerina-platform/persist-tools/pull/54/commits/0b4967f1e318692ce31209fe0643a8f6416b6b63)